mico-core/src/main/java/io/github/ust/mico/core/web/ApplicationController.java

```diff
@@ -110,7 +127,11 @@
             throw new ResponseStatusException(HttpStatus.CONFLICT,
                 "Application '" + newApplication.getShortName() + "' '" + newApplication.getVersion() + "' already exists.");
         }
-        for (MicoService providedService : newApplication.getServices()) {
+        
+        List<MicoService> servicesOfNewApplication = serviceDeploymentInfoRepository
```

Comment:
The new application is not stored in the database yet. Here we just want to get the services that are included in the request body.

Reply:
Nevertheless, why do you use the serviceDeploymentInfoRepository and not the serviceRepository?
Furthermore the corresponding unit tests are not consistent (serviceDeploymentInfoRepository is not (always) mocked).

Reply:
Correct, services now retrieved from deployment info of application in the request body. Unnecessary mocks in corresponding unit tests removed.